frontend/lib/src/components/widgets/DataFrame/hooks/useDataLoader.ts

```diff
@@ -63,15 +64,21 @@ function useDataLoader(
 
       const originalCol = column.indexNumber
       const originalRow = editingState.current.getOriginalRowIndex(row)
-
+      const isAddedRow = editingState.current.isAddedRow(originalRow)
       // Use editing state if editable or if it is an appended row
-      if (column.isEditable || editingState.current.isAddedRow(originalRow)) {
+      if (column.isEditable || isAddedRow) {
         const editedCell = editingState.current.getCell(
           originalCol,
           originalRow
         )
-        if (editedCell !== undefined) {
+        if (notNullOrUndefined(editedCell)) {
           return editedCell
+        } else if (isAddedRow) {
+          // If its an added row, we just return an empty cell here since
+          // since its guaranteed that there will not be a corresponding
```

Comment:
nit: "since since" -> "since"

could add a brief *why* we are returning an empty cell for added rows in the first place? Does this mean if I add a row in the browser, and then I click into the cell of this added row (before its synced to the backend), this will return an empty cell?

Reply:
If you add a new row, all the cells are expected to be empty... so an empty cell is the default state of a cell in an added row. The fallback here is not expected to be called so it shouldn't make any difference. I changed it to return an error cell instead, since I think there isn't a known scenario where this should happen. But its better to have a dedicated fallback here instead of running into the code below which would try to get the cell from arrow.